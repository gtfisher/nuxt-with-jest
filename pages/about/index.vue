<template>
  <div class="container">
    <div>
      <Logo />
      <h1>About</h1>
      <Btn label="click me!" />
      <NuxtLink to="/">Home page</NuxtLink>
    </div>
  </div>
</template>

<script lang="ts">
// import Vue from 'vue'
import Btn from '@/components/Btn/Btn.vue'

export default {
  components: {
    Btn,
  },
}
</script>
