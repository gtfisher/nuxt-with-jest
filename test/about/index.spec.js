import { shallowMount } from '@vue/test-utils'
import about from '../../pages/about/index'

describe('index', () => {
  test('mounts properly', () => {
    const wrapper = shallowMount(about, {
      stubs: ['Logo', 'NuxtLink'],
    })
    expect(wrapper.vm).toBeTruthy()
    expect(wrapper.text()).toContain('About')
  })

  test('renders properly', () => {
    const wrapper = shallowMount(about, {
      stubs: ['Logo', 'NuxtLink'],
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
