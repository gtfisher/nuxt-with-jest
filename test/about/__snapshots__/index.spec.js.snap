// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index renders properly 1`] = `
"<div class=\\"container\\">
  <div>
    <logo-stub></logo-stub>
    <h1>About</h1>
    <btn-stub label=\\"click me!\\"></btn-stub>
    <nuxtlink-stub to=\\"/\\">Home page</nuxtlink-stub>
  </div>
</div>"
`;
